lib/slacks/connection.rb

```diff
@@ -330,7 +330,26 @@ def get_user_id_for_dm(dm)
 
 
     def api(command, params={})
-      response = http.post(command, params.merge(token: token))
+      params_with_token = params.merge(token: token)
+      page_limit = params.key?(:page_limit) ? params.delete(:page_limit) : MAX_PAGES
```

Comment:
It might also be possible to replace line 334 with this method signature?

```ruby
def api(command, page_limit: MAX_PAGES, **params)
```

I think this would require no change of calls like this
```ruby
api("conversations.list", string: "cheese", number: 5)
```

but calls like this
```ruby
params = { string: "cheese", number: 5 }
api("conversations.list", params)
```

would have to change to this, I think:
```diff
params = { string: "cheese", number: 5 }
- api("conversations.list", params)
+ api("conversations.list", **params) 
```

Reply:
I'd be okay with this API change, and the updated calls would be just contained to this file, right? I'll go ahead and make that change then. 👍